frontend/model/contracts/Contract.js

```diff
@@ -0,0 +1,66 @@
+'use strict'
+
+import sbp from '~/shared/sbp.js'
+import { GIMessage } from '~/shared/GIMessage.js'
+
+// TODO: define a flow type for contracts
+export function DefineContract (contract: Object) {
+  const meta = contract.metadata || { validate () {}, create: () => ({}) }
+  sbp('sbp/selectors/register', {
+    [`${contract.name}/create`]: function (data) {
+      const metadata = meta.create()
+      contract.contract.validate(data)
+      meta.validate(metadata)
+      return GIMessage.create(null, null, undefined, `${contract.name}/process`, data, metadata)
```

Comment:
Yes, this is a good idea and something I've given thought to myself. And since you brought it up it gives me more reason to do it. This is especially relevant for #603, which will add new options to the `create` function.

Reply:
Resolved via an edit to #603